JWT/JWT.cs

```diff
@@ -18,17 +20,31 @@ public enum JwtHashAlgorithm
     /// </summary>
     public static class JsonWebToken
     {
-        private static Dictionary<JwtHashAlgorithm, Func<byte[], byte[], byte[]>> HashAlgorithms;
+        private static readonly IDictionary<JwtHashAlgorithm, Func<byte[], string, byte[], byte[]>> HashAlgorithms;
         private static JavaScriptSerializer jsonSerializer = new JavaScriptSerializer();
 
         static JsonWebToken()
         {
-            HashAlgorithms = new Dictionary<JwtHashAlgorithm, Func<byte[], byte[], byte[]>>
+            HashAlgorithms = new Dictionary<JwtHashAlgorithm, Func<byte[], string, byte[], byte[]>>
+            {
+                { JwtHashAlgorithm.HS256, (key, _, value) => { using (var sha = new HMACSHA256(key)) { return sha.ComputeHash(value); } } },
+                { JwtHashAlgorithm.HS384, (key, _, value) => { using (var sha = new HMACSHA384(key)) { return sha.ComputeHash(value); } } },
+                { JwtHashAlgorithm.HS512, (key, _, value) => { using (var sha = new HMACSHA512(key)) { return sha.ComputeHash(value); } } },
+                { JwtHashAlgorithm.RS256, (key, secret, value) => RS256(key, secret, value) }
```

Comment:
Maybe we should separate this into symmetric and asymmetric algorithms, or leave this as-is and treat RS256 as a special case since it needs different arguments.